pkg/scale/decode.go

```diff
@@ -105,15 +113,29 @@ func (d *Decoder) Decode(dst interface{}) (err error) {
 }
 
 // NewDecoder is constructor for Decoder
-func NewDecoder(r io.Reader) (d *Decoder) {
-	d = &Decoder{
-		decodeState{r},
+func NewDecoder(r io.Reader, options ...func(*Decoder)) (d *Decoder) {
+	d = &Decoder{}
+	// Default values...
+	d.decodeState.Reader = r
+	d.decodeState.maxLength = defaultMaxLength
+
+	// Option parameters values:
+	for _, option := range options {
+		option(d)
 	}
 	return
 }
 
+// OptionMaxLength sets the maximum length of the data to be decoded
+func OptionMaxLength(maxLength int) func(*Decoder) {
+	return func(options *Decoder) {
+		options.maxLength = maxLength
+	}
+}
+
 type decodeState struct {
 	io.Reader
+	maxLength int
```

Comment:
That's gonna be a niche nit...

getting the size through file.Stat() in Go returns a `int64`;
maybe we should have this `maxLength` as `int64` if i.e. the writer is a file, and the OS is 32 bit (then `int` = `int32`)